main/src/ca/uwaterloo/flix/language/ast/TokenKind.scala

```diff
@@ -195,6 +195,256 @@ sealed trait TokenKind {
       case TokenKind.Err(_) => "<error>"
     }
   }
+
+  /**
+    * Checks if this token is a line or block comment.
+    */
+  def isCommentNonDoc: Boolean = this match {
+    case TokenKind.CommentLine | TokenKind.CommentBlock => true
+    case _ => false
+  }
+
+  /**
+    * Checks if this token is a doc, line or block comment.
+    */
+  def isComment: Boolean = this == TokenKind.CommentDoc || this.isCommentNonDoc
```

Comment:
The equality check here is costly. Remember it goes into equals which probably checks that the strings are the same!

Reply:
I stand corrected. These are case objects without any data, so should be fine.

Reply:
`this` in this case is a `TokenKind`, which caries no data. I'll change it to all pattern-matches just to be safe though 👍